tests/unit/kernels_reference.cpp

```diff
@@ -174,3 +175,81 @@ std::vector<float> dw::reference::Transpose(const float* in, size_t rows, size_t
 
     return std::move(inT);
 }
+
+void dw::reference::CPUMaxPooling2DForward(const float* input, const deepworks::Shape& shape, float* output, const std::array<int, 2>& kernel,
```

Comment:
Why not a `Tensor` ?

Reply:
Why should there be a tensor? I think all reference implementations should have the same API.

Reply:
New ref implementation take Tensor instead of raw data, we used raw cuz tensor wasn't implemented

Reply:
@l-bat you will use it, your turn